neuro_san/session/session_invocation_context.py

```diff
@@ -44,15 +48,18 @@ def __init__(self, async_session_factory: AsyncAgentSessionFactory,
 
         :param async_session_factory: The AsyncAgentSessionFactory to use
                         when connecting with external agents.
+        :param async_executors_pool: pool of AsyncioExecutors to use for obtaining
+                         an executor instance to use for this context;
         :param llm_factory: The ContextTypeLlmFactory instance
         :param metadata: A grpc metadata of key/value pairs to be inserted into
                          the header. Default is None. Preferred format is a
                          dictionary of string keys to string values.
         """
 
         self.async_session_factory: AsyncAgentSessionFactory = async_session_factory
+        self.async_executors_pool: AsyncioExecutorPool = async_executors_pool
         # Get an async executor to run all tasks for this session instance:
-        self.asyncio_executor: AsyncioExecutor = AsyncioExecutor()
+        self.asyncio_executor: AsyncioExecutor = self.async_executors_pool.get_executor()
         self.origination: Origination = Origination()
```

Comment:
Instead of creating AsynioExecutor instance directly here, request it from the pool.